src/test/ui/suggestions/conversion-methods.stderr

```diff
@@ -0,0 +1,68 @@
+error[E0308]: mismatched types
+  --> $DIR/conversion-methods.rs:15:41
+   |
+15 |     let _tis_an_instants_play: String = "'Tis a fond Ambush—"; //~ ERROR mismatched types
+   |                                         ^^^^^^^^^^^^^^^^^^^^^- help: try using a conversion method: `.to_string()`
+   |                                         |
+   |                                         expected struct `std::string::String`, found reference
+   |
+   = note: expected type `std::string::String`
+              found type `&'static str`
+
+error[E0308]: mismatched types
+  --> $DIR/conversion-methods.rs:16:41
+   |
+16 |     let _just_to_make_bliss: &[isize] = Box::new([1, 2, 3]); //~ ERROR mismatched types
+   |                                         ^^^^^^^^^^^^^^^^^^^- help: try using a conversion method: `.as_slice()`
+   |                                         |
+   |                                         expected &[isize], found struct `std::boxed::Box`
+   |
+   = note: expected type `&[isize]`
+              found type `std::boxed::Box<[{integer}; 3]>`
+
+error[E0308]: mismatched types
+  --> $DIR/conversion-methods.rs:17:46
+   |
+17 |     let _earn_her_own_surprise: &mut [u32] = Box::new([1, 2, 3]); //~ ERROR mismatched types
+   |                                              ^^^^^^^^^^^^^^^^^^^- help: try using a conversion method: `.as_mut_slice()`
+   |                                              |
+   |                                              expected &mut [u32], found struct `std::boxed::Box`
+   |
+   = note: expected type `&mut [u32]`
+              found type `std::boxed::Box<[{integer}; 3]>`
+
+error[E0308]: mismatched types
+  --> $DIR/conversion-methods.rs:18:41
+   |
+18 |     let _but_should_the_play: PathBuf = Path::new("/prv/piercing/earnest");
+   |                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^- help: try using a conversion method: `.to_path_buf()`
+   |                                         |
+   |                                         expected struct `std::path::PathBuf`, found reference
+   |
+   = note: expected type `std::path::PathBuf`
+              found type `&std::path::Path`
+
+error[E0308]: mismatched types
+  --> $DIR/conversion-methods.rs:21:42
+   |
+21 |     let _should_the_glee_glaze: String = 2; // Perhaps surprisingly, we suggest .to_string() here
+   |                                          ^- help: try using a conversion method: `.to_string()`
```

Comment:
Going through all of the examples, it seems to me that the output would be slightly easier to read if the help pointed at the main span, and the suggestion text included the main span's snippet:

```
 --> $DIR/conversion-methods.rs:24:46
   |
24 |     let _in_deaths_stiff_stare: Vec<usize> = &[1, 2, 3]; //~ ERROR mismatched types
   |                                              ^^^^^^^^^^
   |                                              |
   |                                              expected struct `std::vec::Vec`, found reference
   |                                              help: try using a conversion method: `&[1, 2, 3].to_vec()`
```

That being said, I think that can be done in a subsequent PR.